src/mono/mono/component/hot_reload.c

```diff
@@ -1103,17 +1112,47 @@ apply_enclog_pass1 (MonoImage *image_base, MonoImage *image_dmeta, gconstpointer
 }
 
 static void
-set_update_method (MonoImage *image_base, BaselineInfo *base_info, uint32_t generation, MonoImage *image_dmeta, DeltaInfo *delta_info, uint32_t token_index, const char* il_address)
+set_update_method (MonoImage *image_base, BaselineInfo *base_info, uint32_t generation, MonoImage *image_dmeta, DeltaInfo *delta_info, uint32_t token_index, const char* il_address, const char* pdb_address)
 {
 	mono_trace (G_LOG_LEVEL_INFO, MONO_TRACE_METADATA_UPDATE, "setting method 0x%08x in g=%d IL=%p", token_index, generation, (void*)il_address);
 	/* FIXME: this is a race if other threads are doing a lookup. */
 	g_hash_table_insert (base_info->method_table_update, GUINT_TO_POINTER (token_index), GUINT_TO_POINTER (generation));
 	g_hash_table_insert (delta_info->method_table_update, GUINT_TO_POINTER (token_index), (gpointer) il_address);
+	g_hash_table_insert (delta_info->method_ppdb_table_update, GUINT_TO_POINTER (token_index), (gpointer) pdb_address);
+}
+
+static const char *
+hot_reload_get_method_debug_information (MonoImage *image_dppdb, int idx)
+{
+	MonoTableInfo *table_encmap = &image_dppdb->tables [MONO_TABLE_ENCMAP];
+	int rows = table_info_get_rows (table_encmap);
+	for (int i = 0; i < rows ; ++i) {
+		guint32 cols [MONO_ENCMAP_SIZE];
+		mono_metadata_decode_row (table_encmap, i, cols, MONO_ENCMAP_SIZE);
+		int map_token = cols [MONO_ENCMAP_TOKEN];
+		int token_table = mono_metadata_token_table (map_token);
+		if (token_table != MONO_TABLE_METHODBODY)
+			continue;
+		int token_index = mono_metadata_token_index (map_token);
+		if (token_index == idx)
+		{
+			guint32 cols [MONO_METHODBODY_SIZE];
+			MonoTableInfo *methodbody_table = &image_dppdb->tables [MONO_TABLE_METHODBODY];
+			mono_metadata_decode_row (methodbody_table, i, cols, MONO_METHODBODY_SIZE);
+			if (!cols [MONO_METHODBODY_SEQ_POINTS])
+				return NULL;
+
+			const char *ptr = mono_metadata_blob_heap (image_dppdb, cols [MONO_METHODBODY_SEQ_POINTS]);
```

Comment:
I'm a little bit surprised this is a heap offset in the delta pdb.  For the metadata, the offset is in the combined baseline+delta heaps.  (ie if originally the blob heap was 1000 bytes and the delta metadata has 500 more bytes, then the metadata tables in the delta will have offsets like 1001-1500, not 1-500.)

But these formats aren't very consistent.  So I assume you saw in a testcase that it's always offsets in the delta pdb heap only.